src/google/protobuf/util/internal/datapiece.cc

```diff
@@ -305,6 +306,9 @@ StatusOr<int> DataPiece::ToEnum(const google::protobuf::Enum* enum_type,
       value = FindEnumValueByNameWithoutUnderscoreOrNull(enum_type, enum_name);
       if (value != NULL) return value->number();
     }
+
+    // If ignore_unknown_enum_values is true an unknown enum value is treated as 0
+    if (ignore_unknown_enum_values) return 0;
```

Comment:
I think we should return enum_type->enumvalue(0)->number() here. That's the default value you will get when an enum value is not set. In proto3, it's guaranteed to be 0, however, in proto2, it can be something else.

Reply:
fixed